barretenberg/cpp/src/barretenberg/polynomials/row_disabling_polynomial.hpp

```diff
@@ -138,12 +138,23 @@ namespace bb {
  * is equal to \f$ 1-  u_2 \cdots u_{d-1} \f$ otherwise.
  */
 
-template <typename FF> struct RowDisablingPolynomial {
+template <typename Flavor> struct RowDisablingPolynomial {
+    using FF = typename Flavor::FF;
+    static constexpr bool is_ultra_zk = IsAnyOf<Flavor,
+                                                UltraZKFlavor,
+                                                UltraZKRecursiveFlavor_<UltraCircuitBuilder>,
+                                                UltraZKRecursiveFlavor_<MegaCircuitBuilder>>;
     // initialized as a constant linear polynomial = 1
     FF eval_at_0{ 1 };
     FF eval_at_1{ 1 };
 
-    RowDisablingPolynomial() = default;
+    RowDisablingPolynomial()
+    {
+        if constexpr (is_ultra_zk) {
```

Comment:
Not sure if you meant to use different values here but as currently implemented this is redundant with the existing default init to `1` on the members. If you do want case dependent values, probably best to handle both cases within this constructor